### DISCUSSION TITLE
Is that possible to change the application.properties property value in runtime?

### DISCUSSION THREAD

**Discussion by VelSankar-DT** [apache/camel-karavan#1079] [Q&A] [answered]
I like to change the application.properties value in runtime through the rest api. Is that possible to do? or any other alternate solution?

**Comment by davsclaus** [comment]
At runtime in production or during development

**Reply by VelSankar-DT** [reply]
I like to change it at run time in production. I have used the sftp component and stored the input file path in application.properties. Now, I like to read the files from the different locations at run time in production. Is that possible to do Or any other alternative solution for this?

**Reply by davsclaus** [reply, marked as answer]
You can plugin a custom properties source (that can load from a database) then you can use that for this property instead of application.properties.
https://camel.apache.org/manual/using-propertyplaceholder.html#_using_third_party_property_sources

However if you change the value in the database, then you need a way to restart your running Camel application to pickup the change. Or you can restart the pod if you use kubernetes.

**Comment by mgubaidullin** [comment]
You could try env variables

https://camel.apache.org/manual/using-propertyplaceholder.html
